homeassistant/components/media_player/kodi.py

```diff
@@ -157,26 +158,46 @@ def _check_deprecated_turn_off(hass, turn_off_action):
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the Kodi platform."""
     if DATA_KODI not in hass.data:
-        hass.data[DATA_KODI] = []
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    tcp_port = config.get(CONF_TCP_PORT)
-    encryption = config.get(CONF_PROXY_SSL)
-    websocket = config.get(CONF_ENABLE_WEBSOCKET)
-
-    entity = KodiDevice(
-        hass,
-        name=name,
-        host=host, port=port, tcp_port=tcp_port, encryption=encryption,
-        username=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
-        turn_on_action=config.get(CONF_TURN_ON_ACTION),
-        turn_off_action=config.get(CONF_TURN_OFF_ACTION),
-        timeout=config.get(CONF_TIMEOUT), websocket=websocket)
-
-    hass.data[DATA_KODI].append(entity)
-    async_add_devices([entity], update_before_add=True)
+        hass.data[DATA_KODI] = dict()
+
+    # Is this a manual configuration?
```

Comment:
Just check based on if `discovery_info` is `None` or not.

Reply:
This "workflow" was copied from media_player/samsungtv. First I would do it like you say but then thought there is maybe a reason.

Reply:
samsungtv is wrong.